### PR TITLE
new setup documents in English

### PR DESCRIPTION
**Pull request by sherlockblaze**
The remaining two English documents and Chinese documents are modified accordingly.

### CODE REVIEW

File: doc/us-en/setup/Install-OpenEdge-on-Darwin.md [hunk 1]
```diff
@@ -1,2 +1,102 @@
...
+docker version
+```
+
+**For more details, please see the [official documentation](https://docs.docker.com/install/).**
+
+### Install Python2.7 and Python runtime dependency package
+
+> + OpenEdge provides Python Runtime, which supports running code written in Python version 2.7. If you plan to start using ***native*** process mode, you need to install Python 2.7 and the package you depend on. If you plan to start in ***docker*** container mode, you do not need to perform the following steps.
+
+Install by using HomeBrew(Recommened):
```
Comment (ChrisWOW) [inline doc/us-en/setup/Install-OpenEdge-on-Darwin.md]: Recommened?

File: doc/us-en/setup/Install-OpenEdge-on-Darwin.md [hunk 2]
```diff
@@ -1,2 +1,102 @@
...
+
+```shell
+/usr/bin/ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"  // Install HomeBrew
+brew install python@2
+pip isntall protobuf grpcio
```
Comment (ChrisWOW) [inline doc/us-en/setup/Install-OpenEdge-on-Darwin.md]: isntall

File: doc/us-en/setup/Install-OpenEdge-on-Darwin.md [hunk 3]
```diff
@@ -1,2 +1,102 @@
...
+![darwin kernel detail](../../images/setup/os-darwin.png)
+
+Starting OpenEdge containerization mode requires the running device to complete the installation and operation of Docker. You can install it by referring to [Steps above](#Install-Docker).
+
+### Deployment Process
+
+- **Step1**: [Download](../Resources-download.md) OpenEdge archive;
+- **Step2**: Open the terminal and enter the OpenEdge directory for decompression:
+	- execute command `tar -zxvf openedge-xxx.tar.gz`;
+- **Step3**: After the decompression operation is completed, enter the OpenEdge package directory in the terminal, open a new terminal at the same time, execute the command `docker stats`, display the running status of the container in the installted docker, and then execute the command `bin/openedge -w .`, respectively. Observe the contents displayed by the two terminals;
```
Comment (ChrisWOW) [inline doc/us-en/setup/Install-OpenEdge-on-Darwin.md]: installted

Comment (ChrisWOW) [reply doc/us-en/setup/Install-OpenEdge-on-Darwin.md]: ctrl+f 有4个installted

File: doc/us-en/Resources-download.md [hunk 4]
```diff
@@ -1,5 +1,15 @@
 # Resources download
 
+## OpenEdge archieve download
```
Comment (kuailehaibin) [inline doc/us-en/Resources-download.md]: OpenEdge archieve download --> OpenEdge download

File: doc/us-en/setup/Build-OpenEdge-from-Source.md [hunk 5]
```diff
@@ -18,7 +18,8 @@ _**Notice**: OpenEdge requires that the compiled version of Go be above 1.10.0 .
 
 ### Install Docker
 
-> The official Dockerfile is offered for multi-stage builds. If you need to build the relevant image yourself, The version of docker you installed should be above 17.05.
+> + The official Dockerfile is offered for multi-stage builds. If you need to build the relevant image yourself, The version of docker you installed should be above 17.05.
++ According to the [Official Release Log](https://docs.docker.com/engine/release-notes/#18092), the version of docker version < 18.09.2 has some security implications. It is recommended to install/update the Docker version to 18.09. 2 and above.
```
Comment (kuailehaibin) [inline doc/us-en/setup/Build-OpenEdge-from-Source.md]: 18.09. 2 --> 18.09.2

File: doc/us-en/setup/Install-OpenEdge-on-Darwin.md [hunk 6]
```diff
@@ -1,2 +1,102 @@
...
++ The following is an example of the deployment and startup of OpenEdge on Darwin system. It is assumed that the environment required for OpenEdge operation has been configured [before](#Environment-Configuration).
++ The Darwin system mentioned below is based on the Darwin High Sierra Version 10.13.6 with a CPU architecture of x86_64. Execute command `uname -ar` and get the result like this:
+
+![darwin kernel detail](../../images/setup/os-darwin.png)
+
+Starting OpenEdge containerization mode requires the running device to complete the installation and operation of Docker. You can install it by referring to [Steps above](#Install-Docker).
+
+### Deployment Process
+
+- **Step1**: [Download](../Resources-download.md) OpenEdge archive;
```
Comment (kuailehaibin) [inline doc/us-en/setup/Install-OpenEdge-on-Darwin.md]: OpenEdge archive --> OpenEdge

File: doc/us-en/setup/Install-OpenEdge-on-Darwin.md [hunk 7]
```diff
@@ -1,2 +1,102 @@
...
+- **Step2**: Open the terminal and enter the OpenEdge directory for decompression:
+	- execute command `tar -zxvf openedge-xxx.tar.gz`;
```
Comment (kuailehaibin) [inline doc/us-en/setup/Install-OpenEdge-on-Darwin.md]: execute command --> execute the command

File: doc/us-en/setup/Install-OpenEdge-on-Linux.md [hunk 8]
```diff
@@ -1,2 +1,161 @@
...
+
+### Install Docker
+
+> OpenEdge offers two startup modes. To start using ***docker*** container mode (recommended), you need to complete the docker installation first.
+
+**Notice:**
+
++ The official Dockerfile is offered for multi-stage builds. If you need to build the relevant image yourself, The version of docker you installed should be above 17.05.
++ The production environment can run the image using a lower version of Docker, which is currently tested to a minimum usable version of 12.0.
++ According to the [Official Release Log](https://docs.docker.com/engine/release-notes/#18092), the version of docker version < 18.09.2 has some security implications. It is recommended to install/update the Docker version to 18.09. 2 and above.
```
Comment (kuailehaibin) [inline doc/us-en/setup/Install-OpenEdge-on-Linux.md]: the version of docker version < --> the version of docker less than

Comment (kuailehaibin) [reply doc/us-en/setup/Install-OpenEdge-on-Linux.md]: 18.09. 2 --> 18.09.2

File: doc/us-en/setup/Install-OpenEdge-on-Linux.md [hunk 9]
```diff
@@ -1,2 +1,161 @@
...
+curl -sSL https://get.docker.com | sh
+```
+
+#### Ubuntu
+
+Command:
+
+```shell
+sudo snap install docker # Ubuntu16.04 after
+sudo apt install docker.io # Ubuntu 16.04 before
```
Comment (kuailehaibin) [inline doc/us-en/setup/Install-OpenEdge-on-Linux.md]: sudo apt install docker.io --> sudo apt-get install docker.io

File: doc/us-en/setup/Install-OpenEdge-on-Linux.md [hunk 10]
```diff
@@ -1,2 +1,161 @@
...
++ The following is an example of the deployment and startup of OpenEdge on Ubuntu system. The deployment operations on other Linux distributions are basically the same as the following descriptions. It is assumed that the environment required for OpenEdge operation has been configured [before](#Environment-Configuration).
++ The Ubuntu system mentioned below is based on the Linux Ubuntu 4.15.0-29-generic version of the kernel with a CPU architecture of x86_64. Execute command `uname -ar` and get the result like this:
+
+![ubuntu kernel detail](../../images/setup/os-ubuntu.png)
+
+Starting OpenEdge containerization mode requires the running device to complete the installation and operation of Docker. You can install it by referring to [Steps above](#Install-Docker).
+
+### Deployment Process
+
+- **Step1**: [Download](../Resources-download.md) OpenEdge archive;
```
Comment (kuailehaibin) [inline doc/us-en/setup/Install-OpenEdge-on-Linux.md]: OpenEdge archive --> OpenEdge

File: doc/us-en/setup/Install-OpenEdge-on-Linux.md [hunk 11]
```diff
@@ -1,2 +1,161 @@
...
+- **Step2**: Open the terminal and enter the OpenEdge directory for decompression:
+	- execute command `tar -zxvf openedge-xxx.tar.gz`;
```
Comment (kuailehaibin) [inline doc/us-en/setup/Install-OpenEdge-on-Linux.md]: execute command --> execute the command

File: doc/zh-cn/setup/Build-OpenEdge-from-Source.md [hunk 12]
```diff
@@ -18,7 +18,8 @@ _**提示**: OpenEdge要求编译使用的Go版本在 1.10.0 以上。_
 
 ### Docker 安装
 
-> 官方提供 Dockerfile 为多阶段镜像构建，如需自行构建相关镜像，需要安装17.05 及以上版本的 Docker 来build Dockerfile。
+> + 官方提供 Dockerfile 为多阶段镜像构建，如需自行构建相关镜像，需要安装17.05 及以上版本的 Docker 来build Dockerfile。
++ 根据[官方Release日志](https://docs.docker.com/engine/release-notes/#18092)说明，docker version < 18.09.2 的版本具有一些安全隐患，建议安装/更新 Docker 版本到 18.09.2及以上。
```
Comment (kuailehaibin) [inline doc/zh-cn/setup/Build-OpenEdge-from-Source.md]: 根据[官方 Release 日志](https://docs.docker.com/engine/release-notes/#18092)说明，低于 18.09.2 的 Docker 版本具有一些安全隐患，建议安装/更新 Docker 版本到 18.09.2及以上。

File: doc/us-en/setup/Install-OpenEdge-on-Linux.md [hunk 13]
```diff
@@ -1,2 +1,161 @@
...
+
+The bin directory stores the openedge executable binary file, the etc directory stores the configuration of OpenEdge, and the var directory stores the configuration and resources for the modules of OpenEdge.
+
+Then, open a new terminal and execute the command `docker stats` to view the running status of the container in the installed docker, as shown in the following picture:
+
+![view the docker containers status](../../images/setup/docker-stats-before.png)
+
+It can be found that the current system does not have a docker container running.
+
+Then, go to the decompressed OpenEdge folder, execute the command `bin/openedge -w .` in the another terminal, observe the log of OpenEdge startup, as shown below:
```
Comment (kuailehaibin) [inline doc/us-en/setup/Install-OpenEdge-on-Linux.md]: Then, go to the decompressed folder of OpenEdge, execute the command `bin/openedge -w`. In the another terminal, observe the log of OpenEdge startup, as shown below:

File: doc/us-en/setup/Install-OpenEdge-on-Linux.md [hunk 14]
```diff
@@ -1,2 +1,161 @@
...
+
+***Notice:*** The official download page only provides the docker mode executable file. If you want to run in process mode, please refer to [Build-OpenEdge-From-Source.md](./Build-OpenEdge-from-Source.md)
+
+### Start deployment
+
+As mentioned above, download OpenEdge archive from the [Download page](../Resources-download.md) first (also can compile from source, see [Build-OpenEdge-From-Source.md](./Build-OpenEdge-from-Source.md)), then open the terminal to enter OpenEdge directory for decompression. After successful decompression, you can find that the openedge directory mainly includes bin, etc, var, etc., as shown in the following picture:
+
+![OpenEdge directory](../../images/setup/openedge-dir.png)
+
+The bin directory stores the openedge executable binary file, the etc directory stores the configuration of OpenEdge, and the var directory stores the configuration and resources for the modules of OpenEdge.
```
Comment (kuailehaibin) [inline doc/us-en/setup/Install-OpenEdge-on-Linux.md]: The `bin` directory stores the openedge executable binary file, the `etc` directory stores the configuration of OpenEdge, and the `var` directory stores the configuration and resources for the modules of OpenEdge.

File: doc/us-en/setup/Install-OpenEdge-on-Linux.md [hunk 15]
```diff
@@ -1,2 +1,161 @@
...
+	- execute command `tar -zxvf openedge-xxx.tar.gz`;
+- **Step3**: After the decompression operation is completed, enter the OpenEdge package directory in the terminal, open a new terminal at the same time, execute the command `docker stats`, display the running status of the container in the installed docker, and then execute the command `bin/openedge -w .`, respectively. Observe the contents displayed by the two terminals;
+- **Step4**: If the results are consistent, it means that OpenEdge has started normally.
+
+***Notice:*** The official download page only provides the docker mode executable file. If you want to run in process mode, please refer to [Build-OpenEdge-From-Source.md](./Build-OpenEdge-from-Source.md)
+
+### Start deployment
+
+As mentioned above, download OpenEdge archive from the [Download page](../Resources-download.md) first (also can compile from source, see [Build-OpenEdge-From-Source.md](./Build-OpenEdge-from-Source.md)), then open the terminal to enter OpenEdge directory for decompression. After successful decompression, you can find that the openedge directory mainly includes bin, etc, var, etc., as shown in the following picture:
```
Comment (kuailehaibin) [inline doc/us-en/setup/Install-OpenEdge-on-Linux.md]: As mentioned above, download OpenEdge from the [Download page](../Resources-download.md) first (also can compile from source, see [Build-OpenEdge-From-Source.md](./Build-OpenEdge-from-Source.md)), then open the terminal to enter the decompressed directory of OpenEdge. After successfully decompressed, you can find that the OpenEdge directory mainly includes `bin`, `etc`, `var`, etc., as shown in the following picture:

File: doc/us-en/setup/Install-OpenEdge-on-Linux.md [hunk 16]
```diff
@@ -1,2 +1,161 @@
...
+```
+
+## Deploy OpenEdge
+
+### Preparation Before Deployment
+
+**Statement**:
+
++ The following is an example of the deployment and startup of OpenEdge on Ubuntu system. The deployment operations on other Linux distributions are basically the same as the following descriptions. It is assumed that the environment required for OpenEdge operation has been configured [before](#Environment-Configuration).
++ The Ubuntu system mentioned below is based on the Linux Ubuntu 4.15.0-29-generic version of the kernel with a CPU architecture of x86_64. Execute command `uname -ar` and get the result like this:
```
Comment (kuailehaibin) [inline doc/us-en/setup/Install-OpenEdge-on-Linux.md]: The Ubuntu system mentioned below is based on the version of kernel is 4.15.0-29-generic, and the CPU architecture is x86_64. Then execute the command `uname -ar` and get the result like this:

File: doc/us-en/setup/Install-OpenEdge-on-Linux.md [hunk 17]
```diff
@@ -1,2 +1,161 @@
...
+alias python=/yourpath/python2.7
+```
+
+## Deploy OpenEdge
+
+### Preparation Before Deployment
+
+**Statement**:
+
++ The following is an example of the deployment and startup of OpenEdge on Ubuntu system. The deployment operations on other Linux distributions are basically the same as the following descriptions. It is assumed that the environment required for OpenEdge operation has been configured [before](#Environment-Configuration).
```
Comment (kuailehaibin) [inline doc/us-en/setup/Install-OpenEdge-on-Linux.md]: The following is an example of the deployment and startup of OpenEdge on Ubuntu system. The deployment operations on other Linux distributions are basically the same as the following descriptions. It is assumed that the environment required for OpenEdge operation has been [configured](#Environment-Configuration).

File: doc/us-en/setup/Install-OpenEdge-on-Linux.md [hunk 18]
```diff
@@ -1,2 +1,161 @@
...
+curl "https://bootstrap.pypa.io/get-pip.py" -o "get-pip.py"
+python2.7 get-pip.py
+pip install protobuf grpcio
+```
+
+> Enter the command `python -V` to see that the Python version is 2.7.* and the installation is correct.
+
+### Specify the default Python version
+
+In some cases, you need to specify the default Python version for the above installed version. Complete with the following command (Valid after reboot):
```
Comment (kuailehaibin) [inline doc/us-en/setup/Install-OpenEdge-on-Linux.md]: In some cases, you need to specify the default version for Python of the above installed. Specifically, you can execute the commond as below(Valid after reboot).

File: doc/us-en/setup/Install-OpenEdge-on-Linux.md [hunk 19]
```diff
@@ -1,2 +1,161 @@
...
+yum install gcc openssl-devel bzip2-devel
+wget https://www.python.org/ftp/python/2.7.15/Python-2.7.15.tgz
+tar xzf Python-2.7.15.tgz
+make altinstall
+curl "https://bootstrap.pypa.io/get-pip.py" -o "get-pip.py"
+python2.7 get-pip.py
+pip install protobuf grpcio
+```
+
+> Enter the command `python -V` to see that the Python version is 2.7.* and the installation is correct.
```
Comment (kuailehaibin) [inline doc/us-en/setup/Install-OpenEdge-on-Linux.md]: > Execute the command `python -V` to see whether the version of Python is 2.7.*.

File: doc/us-en/setup/Install-OpenEdge-on-Linux.md [hunk 20]
```diff
@@ -1,2 +1,161 @@
...
+
+```shell
+curl -sSL https://get.docker.com | sh
+```
+
+**For more details, please see the [official documentation](https://docs.docker.com/install/).**
+
+### Install Python2.7 and Python runtime dependency package
+
+> + OpenEdge provides Python Runtime, which supports running code written in Python version 2.7. If you plan to start using ***native*** process mode, you need to install Python 2.7 and the package you depend on. If you plan to start in ***docker*** container mode, you do not need to perform the following steps.
```
Comment (kuailehaibin) [inline doc/us-en/setup/Install-OpenEdge-on-Linux.md]: OpenEdge provides Python2.7 runtime, which supports developers write scripts in Python2.7 directly. If you run OpenEdge in **native process mode**, you **MUST** firstly install Python2.7 and the package actually use. But, if you run OpenEdge in **docker container mode**, you do not need to perform the following steps.

File: doc/us-en/setup/Install-OpenEdge-on-Darwin.md [hunk 21]
```diff
@@ -1,2 +1,102 @@
...
+
+Install by using HomeBrew(Recommended):
+
+```shell
+/usr/bin/ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"  // Install HomeBrew
+brew install python@2
+pip install protobuf grpcio
+```
+
+***Notice***: Execute the following command to check the installed Python version:
```
Comment (kuailehaibin) [inline doc/us-en/setup/Install-OpenEdge-on-Darwin.md]: ***Notice***: Execute the following command to check the installed version of Python:

File: doc/us-en/setup/Install-OpenEdge-on-Darwin.md [hunk 22]
```diff
@@ -1,2 +1,102 @@
...
+
+```shell
+python -V
+```
+ 
+Specify the default Python version with following command:
```
[the remark this answers is not in the document]
Comment (sherlockblaze) [reply doc/us-en/setup/Install-OpenEdge-on-Darwin.md]: 这里没必要

File: doc/us-en/setup/Install-OpenEdge-on-Linux.md (hunk 9 above)
Comment (sherlockblaze) [reply doc/us-en/setup/Install-OpenEdge-on-Linux.md]: 这里没必要

**Pull request merged**